File: zoomx/src/main/java/com/zoomx/zoomx/networklogger/NetworkLogManager.java
package com.zoomx.zoomx.networklogger;

import android.os.AsyncTask;

import com.zoomx.zoomx.config.ZoomX;
import com.zoomx.zoomx.model.RequestEntity;

/**
 * Created by Ahmed Fathallah on 1/11/2018.
 */

public class NetworkLogManager {

    public static void log(RequestEntity.Builder builder) {
        if (builder != null)
            new DbAsyncTask().execute(builder);
    }

    public static class DbAsyncTask extends AsyncTask<RequestEntity.Builder, Void, Void> {
        @Override
        protected Void doInBackground(RequestEntity.Builder... builders) {
            ZoomX.getRequestDao().insertRequest(builders[0].create());
            return null;
        }
    }
}

File: zoomx/src/main/java/com/zoomx/zoomx/networklogger/NetworkLogManager.kt
package com.zoomx.zoomx.networklogger

import com.zoomx.zoomx.config.ZoomX
import com.zoomx.zoomx.model.RequestEntity
import io.reactivex.Completable
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers
import timber.log.Timber

/**
 * Created by Ahmed Fathallah on 1/11/2018.
 */

object NetworkLogManager {


    fun log(builder: RequestEntity.Builder) {
        insertRequest(builder).subscribe()
    }

    private fun insertRequest(builder: RequestEntity.Builder): Completable {
        return Completable
                .fromAction { ZoomX.getRequestDao().insertRequest(builder.create()) }
                .doOnError { Timber.e(it) }
                .observeOn(AndroidSchedulers.mainThread())
                .subscribeOn(Schedulers.io())
    }
}
